include/dynd/types/str_util.hpp

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright (C) 2011-15 DyND Developers
+// BSD 2-Clause License, see LICENSE.txt
+//
+
+static inline bool is_simple_identifier_name(const std::string &s)
+{
+  if (s.empty()) {
+    return false;
+  }
+  else {
+    auto c = s.begin();
+    if (!(('a' <= *c && *c <= 'z') || ('A' <= *c && *c <= 'Z') || *c == '_')) {
+      return false;
+    }
+    for (++c; c != s.end(); ++c) {
+      if (!(('0' <= *c && *c <= '9') || ('a' <= *c && *c <= 'z') || ('A' <= *c && *c <= 'Z') || *c == '_')) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
I don't want this extra file, but I agree that `is_simple_identifier_name` needs to go somewhere else. I'm not sure where...